src/proxy/inbound.rs

```diff
@@ -57,25 +61,45 @@ impl Inbound {
         })
     }
 
+    pub(super) async fn start<I>(&self, builder: Builder<I>)
+    where
+        I: Accept,
+        <I as Accept>::Conn: AsyncRead + AsyncWrite + Unpin + Send + 'static,
+        <I as Accept>::Error: std::error::Error + Send + Sync + 'static,
```

Comment:
This is so complex

Reply:
Hi @hzxuzhonghu, yes, there is custom TLS acceptor, so have to declare these traits.

Reply:
One of the goals of this project is to *not* use complex Rust feeatures where possible because ~everyone developing on it is new to rust. IMO this is a regression in the readability of the system. There is nothing so wrong with copying a bit of code.

Reply:
Hi @howardjohn & @hzxuzhonghu, I am also a new to use the Rust, I understand that it's better to make the code readability. However, this is not the most complicated if checking the custom TLS acceptor of `BoringTlsAcceptor` in `src/tls/boring.rs`.  The definition and implementation of `impl<C, F> tls_listener::AsyncTls<C> for BoringTlsAcceptor<F>` is the root cause why I have to add all these traits in this generic implementation.

Reply:
BoringTlsAcceptor is as simple as it can be as far as I can tell. The complexity is required since we are integrating with third party libraries. We don't need such complexity for first party use cases.

Reply:
Yeah, @howardjohn, I understand ztunnel want to avoid the complicated use case of Rust, so we are avoiding the reference usage, however, in frankly, generic pattern with traits bound is widely used and is the most important feature in Rust, including the libs and applications. I hope we can accept it in ztunnel as well.